egs/wsj/s5/steps/libs/nnet3/train/frame_level_objf/common.py

```diff
@@ -79,6 +84,42 @@ def train_new_models(dir, iter, srand, num_jobs,
             # and computation-requests) during training.
             cache_write_opt = "--write-cache={dir}/cache.{iter}".format(
                 dir=dir, iter=iter+1)
+        multitask_egs_opt = common_train_lib.get_multitask_egs_opts(egs_dir, egs_prefix="",
+            archive_index=archive_index,
+            use_multitask_egs=use_multitask_egs)
+
+        if use_multitask_egs:
+            egs_rspecifier = (
+                "ark,bg:nnet3-copy-egs {frame_opts} {context_opts} "
+                "{multitask_egs_opt} "
+                "scp:{egs_dir}/egs.{archive_index}.scp ark:- | "
```

Comment:
The shuffle and merge order is different between these two branches, since the egs in each mini-batch belong to same tasks.